src/DynamoCore/Core/CustomNodeManager.cs

```diff
@@ -440,11 +453,12 @@ private IEnumerable<CustomNodeInfo> ScanNodeHeadersInDirectory(string dir, bool
         }
 
         /// <summary>
-        /// Stores the path and function definition without initializing a node.  Overwrites
-        /// the existing NodeInfo if necessary
+        /// Stores the path and function definition without initializing a node.  
+        /// Overwrites the existing NodeInfo if necessary!
         /// </summary>
-        private void SetNodeInfo(CustomNodeInfo newInfo, bool isTestMode)
+        private void SetNodeInfo(CustomNodeInfo newInfo)
         {
+            //TODO what is this doing?
             var guids = NodeInfos.Where(x =>
```

Comment:
I dont understand this filtering either..? @aparajit-pratap @reddyashish  Any idea?

Reply:
I think it is getting all the nodes from the dict that has the same path as the custom node that is being passed into this SetNodeInfo(). Not sure if this was added to prevent the package from having duplicate nodes.

Reply:
No, this was existing code. I didn't touch it. Not me 🏃

Reply:
This is the reason why it was added: https://github.com/DynamoDS/Dynamo/commit/5eeb1a431d0307568ed83295d74ff5ca037813aa

Reply:
Thanks @aparajit-pratap, I see now. Seems without this, dyfs saving to the same location will end up causing dup nodes in library.